### PR TITLE
GP-51 Update LinkedList tests with reflection

### PR DESCRIPTION
**Pull request by mathodcoast**

### CODE REVIEW

**Review by tboychuk — APPROVED** [review]
@mathodcoast overall everything is good! ✅
* please follow `methodNameDoesSomethingWhenSomeCondition` for test names
* please use get rid of throws declaration in helper methods
Well done with reflection! 😎

File: 2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_list/LinkedListTest.java [hunk 1]
```diff
@@ -306,57 +396,149 @@ void contains() {
...
+            nodeObject = createNodeByConstructorWithoutParameters(element, constructor);
+        }
+        return nodeObject;
+    }
+
+    private Object createNodeByConstructorWithoutParameters(int element, Constructor<?> constructor) throws InstantiationException, IllegalAccessException, java.lang.reflect.InvocationTargetException {
+        Object nodeObject;
+        nodeObject = constructor.newInstance();
+        Field nodeElement = getAccessibleFieldByPredicate(nodeObject, ELEMENT_FIELD);
+        nodeElement.setInt(nodeObject, element);
```
Comment (mathodcoast) [inline 2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_list/LinkedListTest.java:532]: use set() instead setInt()

File: 2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_list/LinkedListTest.java [hunk 2]
```diff
@@ -306,57 +396,149 @@ void contains() {
...
+        sizeField.setInt(intList, size);
+    }
+
+    @SneakyThrows
+    private Object createNodeObjectWithInternalElement(Class<?> nodeClass, int element) {
+        Object nodeObject;
+        Constructor<?>[] declaredConstructors = nodeClass.getDeclaredConstructors();
+        Constructor<?> constructor;
+
+        if (declaredConstructors.length > 0){
```
Comment (mathodcoast) [inline 2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_list/LinkedListTest.java:513]: Redundant logic

File: 2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_list/LinkedListTest.java [hunk 3]
```diff
@@ -1,302 +1,392 @@
...
-        assertThat(intList.size()).isEqualTo(4);
+    void getByZeroIndexWhenListIsEmpty() {
+        assertThatExceptionOfType(IndexOutOfBoundsException.class)
+                .isThrownBy(() -> intList.get(0));
     }
 
     @Test
     @Order(20)
-    void getFirstElementFromEmptyList() {
+    void getWhenIndexIsNegative() {
```
Comment (tboychuk) [inline 2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_list/LinkedListTest.java:266]: @mathodcoast it should be `getThrowsExceptionWhenIndexIsNegative `

File: 2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_list/LinkedListTest.java [hunk 4]
```diff
@@ -1,302 +1,392 @@
...
     @Order(25)
-    void removeLastElement() {
-        intList = LinkedList.of(4, 6, 8, 9);
-
-        int deletedElement = intList.remove(intList.size() - 1);
-
-        assertThat(intList.get(intList.size() - 1)).isEqualTo(8);
-        assertThat(intList.size()).isEqualTo(3);
-        assertThat(deletedElement).isEqualTo(9);
+    void getLastWhenListIsEmpty() {
```
Comment (tboychuk) [inline 2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_list/LinkedListTest.java:309]: @mathodcoast `getLastThrowsExceptionWhenListIsEmpty `

File: 2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_list/LinkedListTest.java [hunk 5]
```diff
@@ -1,302 +1,392 @@
...
+        int size = getInternalSize();
+
+        assertThat(deletedElement).isEqualTo(9);
+        assertThat(newLastElement).isEqualTo(8);
+        assertThat(size).isEqualTo(3);
+    }
+
+    @Test
+    @Order(29)
+    void removeWhenListIsEmpty() {
```
Comment (tboychuk) [inline 2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_list/LinkedListTest.java:363]: @mathodcoast `removeThrowsExceptionWhenListIsEmpty`

**Pull request closed without merging**